File: 14.longest_common_prefix/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) return "";
    var prefix = strs[0];
    for(var i = 1; i < strs.length; i++){
        while(strs[i].indexOf(prefix) !== 0){
            prefix = prefix.substr(0, prefix.length-1);
            if (prefix === '') return "";
        }
    }
    return prefix;
};

File: 14.longest_common_prefix/longestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        prefix = strs[0]
        for i in xrange(len(strs)-1):
            common = ""
            a,b = strs[i], strs[i+1]
            for j in xrange(min(len(a), len(b))):
                if a[j] == b[j]:
                    common += a[j]
                else: break
            prefix = common if len(common) < len(prefix) else prefix
        return prefix